Fix length passed to H265PpsParser::ParsePpsIdFromSliceSegmentLayerRbsp() when called from ParseFuNalu()

ddkilzer opened the pull request:
#### 653fe1d005a464e61ccda68256b5417fdb7382ff
<pre>
Fix length passed to H265PpsParser::ParsePpsIdFromSliceSegmentLayerRbsp() when called from ParseFuNalu()
<a href="https://bugs.webkit.org/show_bug.cgi?id=263935">https://bugs.webkit.org/show_bug.cgi?id=263935</a>
&lt;rdar://117708319&gt;

Reviewed by Youenn Fablet.

* Source/ThirdParty/libwebrtc/Source/webrtc/modules/rtp_rtcp/source/video_rtp_depacketizer_h265.cc:
(webrtc::(anonymous namespace)::ParseFuNalu):
- Compute the correct remaining buffer length after adding an offset to
  the start of the rtp_payload buffer.

Canonical link: <a href="https://commits.webkit.org/270002@main">https://commits.webkit.org/270002@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/966bc04fb1f4514d393c1fda7cfede0c1c1bbf20

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24199 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2308 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25281 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26331 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22282 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3935 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24668 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22737 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24442 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1826 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20909 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26920 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1575 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21828 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28057 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22057 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22123 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25838 "Found 2 new API test failures: /WebKitGTK/TestLoaderClient:/webkit/WebKitURIResponse/http-headers, /WebKitGTK/TestLoaderClient:/webkit/WebKitURIRequest/http-headers (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1515 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19174 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/2237 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1921 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3094 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1879 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xingri commented on Source/ThirdParty/libwebrtc/Source/webrtc/modules/rtp_rtcp/source/video_rtp_depacketizer_h265.cc:
```diff
@@ -287,7 +287,7 @@ absl::optional<VideoRtpDepacketizer::ParsedRtpPayload> ParseFuNalu(
         H265PpsParser::ParsePpsIdFromSliceSegmentLayerRbsp(
             rtp_payload.cdata() + kHevcNalHeaderSize + kHevcFuHeaderSize,
-            rtp_payload.size() - kHevcFuHeaderSize, nalu.type);
+            rtp_payload.size() - kHevcNalHeaderSize - kHevcFuHeaderSize, nalu.type);
```
@ddkilzer Thanks for the patch. I will make sure if there are any other missing by reviewing the latest webrtc review 
<https://webrtc-review.googlesource.com/c/src/+/321102>.

ddkilzer replied:
Thanks!

Commit:
Fix length passed to H265PpsParser::ParsePpsIdFromSliceSegmentLayerRbsp() when called from ParseFuNalu()
https://bugs.webkit.org/show_bug.cgi?id=263935
<rdar://117708319>

Reviewed by Youenn Fablet.

* Source/ThirdParty/libwebrtc/Source/webrtc/modules/rtp_rtcp/source/video_rtp_depacketizer_h265.cc:
(webrtc::(anonymous namespace)::ParseFuNalu):
- Compute the correct remaining buffer length after adding an offset to
  the start of the rtp_payload buffer.

Canonical link: https://commits.webkit.org/270002@main
